cmd/proxy-client/main.go

```diff
@@ -101,10 +101,17 @@ func runClient(cCtx *cli.Context) error {
 		return errors.New("invalid combination of --verify-tls and --server-attestation-type passed (only 'none' is allowed)")
 	}
 
+	// Auto-detect client attestation type if not specified
 	clientAttestationType, err := proxy.ParseAttestationType(cCtx.String("client-attestation-type"))
```

Comment:
this function will fail if `cCtx.String("client-attestation-type")` is empty

Reply:
Yes, that's the idea. If parsing fails and client-attestation-type is empty it will try to autodetect.

Reply:
but it will never proceed to line 108, so the new code will never be executed 🤔

Reply:
if client-attestation-type is empty (by default that's ""), then ParseAttestationType will return an err, thus entering the if condition at 106. Checking the string at 108 will return true and enter that statement.